server/scanner/LibraryScanner.js

```diff
@@ -603,3 +617,55 @@ function hasAudioFiles(fileUpdateGroup, itemDir) {
 function isSingleMediaFile(fileUpdateGroup, itemDir) {
   return itemDir === fileUpdateGroup[itemDir]
 }
+
+async function findLibraryItemByItemToItemInoMatch(libraryId, fullPath) {
+  const ino = await fileUtils.getIno(fullPath)
+  if (!ino) return null
+  const existingLibraryItem = await Database.libraryItemModel.findOneOld({
+    libraryId: libraryId,
+    ino: ino
+  })
+  if (existingLibraryItem)
+    Logger.debug(`[LibraryScanner] Found library item with matching inode "${ino}" at path "${existingLibraryItem.path}"`)
+  return existingLibraryItem
+}
+
+async function findLibraryItemByItemToFileInoMatch(libraryId, fullPath, isSingleMedia) {
+  if (!isSingleMedia) return null
+  // check if it was moved from another folder by comparing the ino to the library files
+  const ino = await fileUtils.getIno(fullPath)
+  if (!ino) return null
+  const existingLibraryItem = await Database.libraryItemModel.findOneOld([
+    {
+      libraryId: libraryId
+    },
+    sequelize.where(sequelize.literal('(SELECT count(*) FROM json_each(libraryFiles) WHERE json_valid(json_each.value) AND json_each.value->>"$.ino" = :inode)'), {
+      [sequelize.Op.gt]: 0
+    })
+  ], {
+    inode: ino
+  })
```

Comment:
Some file systems have inode values that are only a few digits long so I added this additional query that uses sqlite json functions. Unfortunately this is an extra query and is sqlite specific but in Sequelize v7 they are adding support for handling JSON queries.